app/components/Layout/Header.tsx

```diff
@@ -80,19 +92,64 @@ const HeaderLayout = ({
             {isLoggedIn ? (
               <>
                 {isRegistered && (
-                  <li>
-                    <Link href="/user/profile">
-                      <a className="nav-button">Profile</a>
-                    </Link>
-                  </li>
+                  <>
+                    <li className="d-none d-lg-block">
+                      <Dropdown alignRight>
+                        <Dropdown.Toggle variant="primary">
+                          <FontAwesomeIcon
+                            color="white"
+                            icon={faUser}
+                            size="1x"
+                          />
+                        </Dropdown.Toggle>
+                        <Dropdown.Menu>
+                          <Dropdown.Item as={Link} href="/user/profile">
+                            <a className="dropdown-item text-right">
+                              <div>
+                                <span>{user.firstName}</span>{' '}
+                                <span>{user.lastName}</span>
+                              </div>
+                              <div className="small text-muted text-nowrap">
+                                {user.emailAddress}
+                              </div>
+                            </a>
+                          </Dropdown.Item>
+                          <Dropdown.Item
+                            as={Form}
+                            action="/logout"
+                            method="post"
+                            tabIndex={0}
+                            onKeyPress={submitForm}
+                          >
+                            <button type="submit" className="w-100 text-right">
+                              Logout
+                            </button>
+                          </Dropdown.Item>
+                        </Dropdown.Menu>
+                      </Dropdown>
+                    </li>
+                    <li className="d-lg-none">
+                      <Link href="/user/profile">
+                        <a className="nav-button text-right">
+                          <div>
+                            <span>{user.firstName}</span>{' '}
+                            <span>{user.lastName}</span>
+                          </div>
+                          <div className="small text-nowrap">
+                            {user.emailAddress}
+                          </div>
+                        </a>
+                      </Link>
+                    </li>
+                    <li className="d-lg-none">
+                      <Form action="/logout" method="post">
+                        <button type="submit" className="nav-button text-right">
+                          Logout
+                        </button>
+                      </Form>
+                    </li>
+                  </>
```

Comment:
I would extract all of this in its own container, which should be a relay `fragmentContainer`, so that the relay fragment fetching the user info is colocated with its rendering, preventing the warning I mentioned in my other comment.

Instead of a `user` prop, the `Header` component would have a `userProfileDropdown` prop, and the `UserProfileDropdown` component would be used in `default-layout`.

Reply:
Ah ok, yes I need to get used to Relay. I don't think I noticed a warning tbh. So you're saying here that `userProfileDropdown` is a prop that would pass down a component instead? I'll try that out...

Reply:
That warning may be at the very top of the file, as there is (was? It seemed to have been fixed with the PR that updated relay things) a bug with the linter